File: R/8kyu/8kyu-beginner-lost-without-a-map.r
# Beginner - Lost Without a Map
# Language: R
# Difficulty: 8 kyu

maps <- function(v) {
  v * 2
}


# Tests
expect_equal(maps(c(1, 2, 3)), c(2, 4, 6))
expect_equal(maps(c(0, 1, 2, 3, 4, 5, 6, 7, 8, 9)), c(0, 2, 4, 6, 8, 10, 12, 14, 16, 18))
expect_equal(maps(numeric()), numeric())

File: Julia/8kyu/8kyu-beginner-lost-without-a-map.jl
# Beginner - Lost Without a Map
# Language: Julia
# Version: 1.5
# Difficulty: 8 kyu

module Solution
export maps

function maps(x)
  x * 2
end

end


# Tests
using FactCheck

facts("Example tests") do
    @fact Solution.maps([1, 2, 3]) --> [2, 4, 6]
    @fact Solution.maps([4, 1, 1, 1, 4]) --> [8, 2, 2, 2, 8]
    @fact Solution.maps([2, 2, 2, 2, 2, 2]) --> [4, 4, 4, 4, 4, 4]
end